pkg/store/did/store.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package did
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+const (
+	// NameSpace for did store
+	NameSpace = "did"
+
+	didNameKey        = "didname_"
+	didNameKeyPattern = didNameKey + "%s"
+
+	// limitPattern for the iterator
+	limitPattern = "%s~"
+)
+
+// ErrDIDNotFound signals that the entry for the given DID and key is not present in the store.
+var ErrDIDNotFound = errors.New("did not found under given key")
+
+// Store stores did doc
+type Store struct {
+	store storage.Store
+}
+
+type provider interface {
+	StorageProvider() storage.Provider
+}
+
+// New returns a new did store
+func New(ctx provider) (*Store, error) {
+	store, err := ctx.StorageProvider().OpenStore(NameSpace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open did store: %w", err)
+	}
+
+	return &Store{store: store}, nil
+}
+
+// SaveDID saves a did doc.
+func (s *Store) SaveDID(name string, didDoc *did.Doc) error {
+	if name == "" {
+		return errors.New("did name is mandatory")
+	}
+
+	id, err := s.GetDIDByName(name)
+	if err != nil && !errors.Is(err, storage.ErrDataNotFound) {
+		return fmt.Errorf("get did id using name : %w", err)
+	}
+
+	if id != "" {
+		return errors.New("did name already exists")
+	}
+
+	docBytes, err := didDoc.JSONBytes()
+	if err != nil {
+		return fmt.Errorf("failed to marshal didDoc: %w", err)
+	}
+
+	if err := s.store.Put(didDoc.ID, docBytes); err != nil {
```

Comment:
does it mean, someone can save DID by name and later also retrieve by ID?

Reply:
yes you can retrieve by both name and id